package/MDAnalysis/analysis/Aggregation_size.py

```diff
@@ -0,0 +1,370 @@
+                if check1 == 1:
+                    restart = True
+                    break
+        # make a dictionary with the key of number of molecules in clusters
+        # and value of number of cluster
+        size_agg = [0] * (len(aggregate)-1)
+        for counter in range(len(aggregate)-1):
+            size_agg[counter] = len(aggregate[counter])
+        self.size_num[self._frame_index] = {
+            j: size_agg.count(j) for j in size_agg}
+        if 0 in self.size_num[self._frame_index].keys():
+            del self.size_num[self._frame_index][0]
+        # Give option to not considering the monomer in calculation.
+        # the default is not considering monomers
+        if self._monomer:
+            if 1 in self.size_num[self._frame_index].keys():
+                del self.size_num[self._frame_index][1]
+        # Give option to calculate average cluster gyration radius thoughout
+        # the simulation. The default is No
+        print("first", self._Gyr_calc)
+        if self._Gyr_calc:
+            Gyr = []
+            for x in range(len(aggregate)-1):
+                aggregate[x] = [
+                    int(elem) for elem in aggregate[x]]
+                if len(aggregate[x]) != 1:
+                    values2 = ([f'resid {i}' for i in aggregate[x]])
+                    iio = u.select_atoms(values2[0])
+                    for x in values2[1:]:
+                        io = (u.select_atoms(x))
+                        iio = iio + io
+                    Gyr.append(iio.radius_of_gyration())
+            Gyr = np.array(Gyr)
+            GyR = np.average(Gyr)
+            self._GYR.append(GyR)
+        self.results[self._frame_index, 0] = self._trajectory.time
+
+    def _conclude(self):
+        if self._Gyr_calc:
+            self.results[:, 4] = np.array(self._GYR)
+        number_average = []
+        weight_average = []
+        z_average = []
+        for counter in range(0, len(self.size_num)):
+            size = list(self.size_num[counter].keys())
+            num = list(self.size_num[counter].values())
+            # number_average
+            number_average.append(np.sum(np.array(num)*(np.array(size))) /
+                                  np.sum(np.array(num)))
+            # weight_average
+            weight_average.append(np.sum(np.array(num)*(np.array(size)**2)) /
+                                  np.sum(np.array(num)*(np.array(size))))
+            # z_average
+            z_average.append(np.sum(np.array(num)*(np.array(size)**3)) /
+                             np.sum(np.array(num)*(np.array(size)**2)))
+        self.results[:, 1] = np.array(number_average)
+        self.results[:, 2] = np.array(weight_average)
+        self.results[:, 3] = np.array(z_average)
+        columns = ['Frame', 'number_average',
+                   'weight_average', 'z_average',
+                   'Radius of Gyration']
+        self.df = pd.DataFrame(self.results, columns=columns)
```

Comment:
Instead of a data frame, consider saving your results as different attributes. We don't have [pandas as a core dependency](https://userguide.mdanalysis.org/1.0.0/module_imports.html#module-dependencies-in-the-code) so it's not guaranteed to be installed in the user's Python environment.

Reply:
results will be reported in self.results with 5 columns. Columns: 1- Frame number 2- number average 3- weight average 4- z average 5- Average Gyration radius. So I can delete the lines which are related to panda